esmvaltool/recipes/recipe_ecs_scatter.yml

```diff
@@ -0,0 +1,411 @@
+          - {dataset: inmcm4, ensemble: r1i1p1}
+          - {dataset: IPSL-CM5A-LR, ensemble: r1i1p1}
+          - {dataset: IPSL-CM5B-LR, ensemble: r1i1p1}
+          - {dataset: MIROC5, ensemble: r1i1p1}
+          - {dataset: MIROC-ESM, ensemble: r1i1p1}
+          - {dataset: MPI-ESM-LR, ensemble: r1i1p1}
+          # - {dataset: MPI-ESM-MR, ensemble: r1i1p1}
+          # - {dataset: MPI-ESM-P, ensemble: r1i1p1}
+          - {dataset: MRI-CGCM3, ensemble: r1i1p1}
+          - {dataset: NorESM1-M, ensemble: r1i1p1}
+    scripts:
+      ecs_scatter:
+        diag: shhc
+        calcmm: true
+        ecs_file: $diag_scripts/emergent_constraints/ecs_cmip.nc
+        legend_outside: false
+        predef_minmax: true
+        script: emergent_constraints/ecs_scatter.ncl
+
+  itczidx:
+    description: southern ITCZ index
+    themes:
+      - EC
+    realms:
+      - atmos
+    variables:
+      pr:
+        reference_dataset: TRMM-L3
+        mip: Amon
+        exp: historical
+        start_year: 1986
+        end_year: 2005
+        project: CMIP5
+        additional_datasets:
+          - {dataset: TRMM-L3, project: obs4mips, level: v7, version: 7A,
+             start_year: 1998, end_year: 2013, tier: 1}
+          - {dataset: ACCESS1-0, ensemble: r1i1p1}
+          # - {dataset: ACCESS1-3, ensemble: r1i1p1}
+          - {dataset: bcc-csm1-1, ensemble: r1i1p1}
+          - {dataset: bcc-csm1-1-m, ensemble: r1i1p1}
+          - {dataset: BNU-ESM, ensemble: r1i1p1}
+          - {dataset: CanESM2, ensemble: r1i1p1}
+          - {dataset: CCSM4, ensemble: r1i1p1}
+          - {dataset: CNRM-CM5, ensemble: r1i1p1}
+          - {dataset: CSIRO-Mk3-6-0, ensemble: r1i1p1}
+          # - {dataset: FGOALS-g2, ensemble: r1i1p1}
+          - {dataset: GFDL-CM3, ensemble: r1i1p1}
+          - {dataset: GFDL-ESM2G, ensemble: r1i1p1}
+          - {dataset: GFDL-ESM2M, ensemble: r1i1p1}
+          - {dataset: GISS-E2-H, ensemble: r1i1p1}
+          - {dataset: GISS-E2-R, ensemble: r1i1p1}
+          - {dataset: HadGEM2-ES, ensemble: r1i1p1}
+          - {dataset: inmcm4, ensemble: r1i1p1}
+          - {dataset: IPSL-CM5A-LR, ensemble: r1i1p1}
+          - {dataset: IPSL-CM5B-LR, ensemble: r1i1p1}
+          - {dataset: MIROC5, ensemble: r1i1p1}
+          - {dataset: MIROC-ESM, ensemble: r1i1p1}
+          - {dataset: MPI-ESM-LR, ensemble: r1i1p1}
+          # - {dataset: MPI-ESM-MR, ensemble: r1i1p1}
+          # - {dataset: MPI-ESM-P, ensemble: r1i1p1}
+          - {dataset: MRI-CGCM3, ensemble: r1i1p1}
+          - {dataset: NorESM1-M, ensemble: r1i1p1}
+      prStderr:
+        reference_dataset: TRMM-L3
+        mip: Amon
+        frequency: mon
+        additional_datasets:
+          - {dataset: TRMM-L3, project: obs4mips, level: v7, version: 7A,
+             start_year: 1998, end_year: 2013, tier: 1}
+    scripts:
+      ecs_scatter:
+        diag: itczidx
+        calcmm: true
+        ecs_file: $diag_scripts/emergent_constraints/ecs_cmip.nc
```

Comment:
If you do not want to include the values in the recipe, I would recommend the following:
- if `ecs_file: null`, use the default values from the installed file
- If a file is specified and the path is relative, it should be relative to `diag_script_info@auxiliary_data_dir`
Using the $diag_scripts environment variable here is really not clear to people not intimately familiar with NCL diagnostics, so that makes it confusing and not user friendly.